src/main/java/org/mitre/dsmiley/httpproxy/ProxyServlet.java

```diff
@@ -453,7 +477,8 @@ protected void copyProxyCookie(HttpServletRequest servletRequest,
       Cookie servletCookie = new Cookie(proxyCookieName, cookie.getValue());
       servletCookie.setComment(cookie.getComment());
       servletCookie.setMaxAge((int) cookie.getMaxAge());
-      servletCookie.setPath(path); //set to the path of the proxy servlet
+      servletCookie.setPath(cookie.getPath());
```

Comment:
Ok, I did not see I break the tests. The problem is when the targeted proxy app set a cookie path different of the cookie path of the proxy webapp. In the proxy webapp you can set the cookie path in the web.xml (Servlet 3.0) or weblogic.xml (for exemple if you this server). 
So set the cookie path starting with the servlet context root path can be an issue in some case, if it's overriden by web.xml config (ex : app context root "myApp" and cookie path "mySuperApp".
In my case I prefere reverse proxy the original cookie path.
Maybe add an initParam to control this ?

web.wml : 
<session-config>
    <cookie-config>
        <name>PROXYWEB_JSESSIONID</name>
        <path>/myApp</path>
    </cookie-config>
</session-config>